loki-logback-appender/src/test/java/com/github/loki4j/logback/JsonLayoutTest.java

```diff
@@ -97,10 +103,76 @@ protected void writeExactlyOneField(JsonEventWriter writer, ILoggingEvent event)
         layout.stop();
     }
 
-    private <T extends JsonProvider<?>> T disable(Supplier<T> ctor) {
-        var provider = ctor.get();
+    @Test
+    public void testThreadSafety() throws Exception {
+        class Result {
+            final String expected;
+            final String actual;
+
+            public Result(final String expected, final String actual) {
+                this.expected = expected;
+                this.actual = actual;
+            }
+        }
+
+        final List<Result> results = Collections.synchronizedList(new ArrayList<>());
+
+        final var layout = jsonMsgLayout();
+        try {
+            layout.setLogLevel(disable(LogLevelJsonProvider::new));
+            layout.setLoggerName(disable(LoggerNameJsonProvider::new));
+            layout.setTimestamp(disable(TimestampJsonProvider::new));
+
+            layout.start();
+
+            // ready...
+            final CountDownLatch latch = new CountDownLatch(1);
+            final ExecutorService executorService = Executors.newFixedThreadPool(16);
+            // set...
+            for (int i = 0; i < 128; i++) {
+                final int threadId = i;
+                executorService.submit(() -> {
+                    awaitLatch(latch);
+                    final var event = loggingEvent(
```

Comment:
Could you please use a generator for log events? You can find an example here: https://github.com/loki4j/loki-logback-appender/blob/main/loki-logback-appender/src/test/java/com/github/loki4j/logback/integration/BatchSizeTest.java#L40

Reply:
Can do. What exactly do you have in mind? Since I need to know how the rendered version of the log event will look like, I would need to render the log events with the same or a different layout instance. Since we are specifically testing that rendering a logging event produces the expected output, this becomes a little bit of a catch-22. Maybe I have misunderstood, so I appreciate if you can clarify a bit. Thanks!

Reply:
If I got your question right, you can simply use getters from `ILoggingEvent`, e.g. `getThreadName()` and `getMessage()` to render the expected version.